crates/ty_python_semantic/src/types/class.rs

```diff
@@ -1972,7 +1974,8 @@ impl<'db> KnownClass {
             | Self::UnionType
             | Self::GeneratorType
             | Self::AsyncGeneratorType
-            | Self::MethodWrapperType => Truthiness::AlwaysTrue,
+            | Self::MethodWrapperType
+            | Self::Field => Truthiness::AlwaysTrue,
```

Comment:
This should go in the third branch (the `Truthiness::Ambiguous` branch) because `Field` is not declared as a `@final` class. See the comment at the top of this method.

Reply:
So, `Field` itself does not implement `__bool__`, and should by truthy, but a subclass of `Field` can implement `__bool__` and have it return False -- is that why we need `@final`?

Reply:
Exactly! If we said that the truthiness of `Field` was always true, then this would imply that all instances of `Field` _and all instances of possible `Field` subclasses_ are all guaranteed to be truthy. This isn't necessarily the case